Error running patch evaluator

When trying to execute the ecaluator for a patch seems like the program can't find the test file correctly. 

From some debugging seems like the error is in this line of code of the function get_test_file() that uses the contract_path to get the test.

https://github.com/ASSERT-KTH/sb-heists/blob/ef74319c8fbef7a74c97d946497e6b21d606d165/evaluator/src/testing/hardhat_runner.py#L24




### For reproduction:

My command line for using evaluator following readme instructions: 

`python src/main.py  --format solidity --patch ~/Documents/RepairComp/results/smartbugs/SolGPT/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e_3round.sol --contract-file ~/Documents/sb-heists/smartbugs-curated/0.4.x/contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e.sol --main-contract PoCGame`

### Logs:
```
Executing command:  npx hardhat compile
Compiled 1 Solidity file successfully (evm target: unknown evm version for solc version 0.4.25).

contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e.sol:201:9: Warning: Return value of low-level calls not used.
        whale.call.value(amount)(bytes4(keccak256("donate()")));
        ^-----------------------------------------------------^

contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e.sol:213:9: Warning: Return value of low-level calls not used.
        whale.call.value(amount)(bytes4(keccak256("donate()")));
        ^-----------------------------------------------------^


Backing up file /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e.sol to ../smartbugs-curated/0.4.x/backups
Executing command:  npx hardhat compile
Compiled 1 Solidity file successfully (evm target: unknown evm version for solc version 0.4.25).


Executing command:  npx hardhat test /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e_test.js


An unexpected error occurred:
Error: Cannot find module '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e_test.js'
Require stack:
- /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/mocha/lib/mocha.js
- /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/mocha/index.js
- /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/builtin-tasks/test.js
- /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/core/tasks/builtin-tasks.js
- /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/core/config/config-loading.js
- /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/cli/cli.js
- /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/cli/bootstrap.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
    at Function.Module._load (node:internal/modules/cjs/loader:975:27)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/mocha/lib/mocha.js:414:36
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/mocha/lib/mocha.js:411:14)
    at Mocha.run (/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/mocha/lib/mocha.js:972:10)
    at testFailures (/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/src/builtin-tasks/test.ts:136:15)
    at new Promise (<anonymous>) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/mocha/lib/mocha.js',
    '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/mocha/index.js',
    '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/builtin-tasks/test.js',
    '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/core/tasks/builtin-tasks.js',
    '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/core/config/config-loading.js',
    '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/cli/cli.js',
    '/home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/node_modules/hardhat/internal/cli/bootstrap.js'
  ]
}


Evaluation Results:
Contract File: /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e.sol
Patch File: unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e_3round.sol
Total Tests: 2
Passed Tests: 2
Sanity Success: 1
Sanity Failures: 0
```

Seems like when searching for the test it does on the wrong folder. See the logs :

`Executing command:  npx hardhat test /home/sofia/Documents/sb-heists/smartbugs-curated/0.4.x/contracts/dataset/unchecked_low_level_calls/0x07f7ecb66d788ab01dc93b9b71a88401de7d0f2e_test.js`

--- a/evaluator/src/testing/hardhat_runner.py
+++ b/evaluator/src/testing/hardhat_runner.py
@@ -21,8 +21,9 @@ class HardhatTestRunner:
         self.compile()
     def get_test_file(self, contract_path: str) -> str:
         """Get the test file for a specific contract"""
-        test_path=contract_path.replace("contracts/dataset", "test")
-        return test_path.replace(".sol", "_test.js")
+        contract_dir, contract_name = os.path.split(contract_path)
+        test_name = contract_name.replace(".sol", "_test.js")
+        return os.path.join("test", os.path.basename(contract_dir), test_name)
 
     def _run_command(self, command: str) -> subprocess.CompletedProcess:
         """Helper method to run shell commands"""

--- a/evaluator/src/testing/hardhat_runner.py
+++ b/evaluator/src/testing/hardhat_runner.py
@@ -20,9 +20,8 @@ class HardhatTestRunner:
         self.hardhat_path = 'npx hardhat'
         self.compile()
     def get_test_file(self, contract_path: str) -> str:
-        print("contract_path", contract_path)
-        test_path=contract_path.replace("contracts/dataset", "test")
         """Get the test file for a specific contract"""
+        test_path=contract_path.replace("contracts/dataset", "test")
         return test_path.replace(".sol", "_test.js")
 
     def _run_command(self, command: str) -> subprocess.CompletedProcess:

--- a/evaluator/src/testing/hardhat_runner.py
+++ b/evaluator/src/testing/hardhat_runner.py
@@ -20,8 +20,10 @@ class HardhatTestRunner:
         self.hardhat_path = 'npx hardhat'
         self.compile()
     def get_test_file(self, contract_path: str) -> str:
+        print("contract_path", contract_path)
+        test_path=contract_path.replace("contracts/dataset", "test")
         """Get the test file for a specific contract"""
-        return os.path.join("test", contract_path.replace(".sol", "_test.js"))
+        return test_path.replace(".sol", "_test.js")
 
     def _run_command(self, command: str) -> subprocess.CompletedProcess:
         """Helper method to run shell commands"""

